### PR TITLE
Issue #6606: catch all exceptions during reflection in AbstractTyeAware

### PR DESCRIPTION
**Pull request by rnveach**
Issue #6606

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractTypeAwareCheck.java [hunk 1]
```diff
@@ -240,7 +240,9 @@ private ClassResolver getClassResolver() {
         try {
             clazz = getClassResolver().resolve(resolvableClassName, className);
         }
-        catch (final ClassNotFoundException ignored) {
+        // -@cs[IllegalCatch] We need to catch all exceptions.
+        // See https://github.com/checkstyle/checkstyle/issues/6606
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractTypeAwareCheck.java]: no need to reference issue, it is already in git history.

lets make comment `exception type is not predictable`

**Comment by pbludov** [comment]
@rnveach please fix the typo in the commit message. Once it comes to release notes, the spell checker will complain.

**Comment by rnveach** [comment]
> Once it comes to release notes, the spell checker will complain.

Release notes is generated from issue title, not commit message. https://github.com/checkstyle/contribution/blob/master/releasenotes-builder/src/main/resources/com/github/checkstyle/templates/xdoc_freemarker.template#L9
Release notes are not spell checked. https://github.com/checkstyle/checkstyle/blob/master/.ci/jsoref-spellchecker/exclude.pl#L13-L14

So it won't be an issue.

> please fix the typo in the commit message

Will do.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractTypeAwareCheck.java (hunk 1 above)
Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractTypeAwareCheck.java]: Done.

**Pull request merged**